File: Java/test/sorts/SelectionSortTest.java
package sorts;

import org.junit.Assert;
import org.junit.Test;
import util.SortUtil;
import util.TestUtil;

/**
 * Created by Chris Keyser on 10/3/2016.
 */
public class SelectionSortTest {
    @Test
    public void SelectionSort_SortsCorrectly() {
        int arr[] = TestUtil.generateRandomArray(TestUtil.ARR_SIZE);

        int[] sorted = SelectionSort.sort(arr);

        Assert.assertTrue(SortUtil.isSorted(sorted));
    }
}

File: Scala/src/test/sorts/SelectionSortTest.scala
package sorts

import com.ckeyser.sorts.SelectionSort
import com.ckeyser.util.SortUtil
import org.scalatest.FunSpec
import util.TestUtil

/**
  * Created by Chris on 10/2/2016.
  */
class SelectionSortTest extends FunSpec {
    it("should correctly sort an array") {
        val arr = TestUtil.RandomArray(TestUtil.ARR_SIZE)

        val result = SelectionSort.sort(arr)

        assert(SortUtil.isSorted(result))
    }
}
